src/components/Toggle/Toggle.props.ts

```diff
@@ -0,0 +1,15 @@
+import { MouseEvent } from 'react';
+
+import type { ToggleConfig } from './Toggle.styles';
+import { HelperTextProps } from '../HelperText';
+
+export type ToggleProps = {
+  isIndeterminate?: boolean;
+  isChecked?: boolean;
+  size?: 'small' | 'large';
+  state?: 'disabled' | undefined;
+  label?: string;
+  helperText?: Pick<HelperTextProps, 'text'>;
+  custom?: ToggleConfig;
+  onChange?: (e?: MouseEvent<HTMLSpanElement>) => void;
+};
```

Comment:
`Toggle` should support all the same events and props as `Checkbox` component, its the same business logic, but  different UI ;)